src/celengine/meshmanager.cpp

```diff
@@ -301,26 +300,21 @@ ConvertTriangleMesh(const M3DTriangleMesh& mesh,
             Eigen::Vector3f position = mesh.getVertex(triVert[j]);
             Eigen::Vector3f normal = vertexNormals[i * 3 + j];
 
-            int dataOffset = (i * 3 + j) * floatsPerVertex;
-            vertexData[dataOffset + 0] = position.x();
-            vertexData[dataOffset + 1] = position.y();
-            vertexData[dataOffset + 2] = position.z();
-            vertexData[dataOffset + 3] = normal.x();
-            vertexData[dataOffset + 4] = normal.y();
-            vertexData[dataOffset + 5] = normal.z();
+            int dataOffset = (i * 3 + j) * vertexSize;
+            std::memcpy(vertexData.data() + dataOffset, position.data(), sizeof(float) * 3);
```

Comment:
won't be std::copy more correct here?

Reply:
vertexData is a set of words of (an alias of) `std::uint32_t`, we need it to contain the bit representation of the floats but copy would try to cast the value.